packages/cosmo/package.py

```diff
@@ -471,10 +463,12 @@ def install(self, spec, prefix):
     @run_after('install')
     @on_package_attributes(run_tests=True)
     def test(self):
+        with open('spec.yaml', mode='w') as f:
+            f.write(self.spec.to_yaml())
         try:
             subprocess.run([
                 self.build_directory + '/test/tools/test_cosmo.py', '-s',
-                str(self.spec), '-b',
+                'spec.yaml', '-b',
```

Comment:
Isn't this an API breaking change, requiring a change in `test_cosmo.py` in all branches/versions and breaking compatibility to `cosmo@5.09a.mch1.2.p2`?
I think [this line](https://github.com/COSMO-ORG/cosmo/blob/master/cosmo/ACC/test/tools/test_cosmo.py#L19) won't be happy with `Spec('spec.yaml')`.

Reply:
You are right. This requires a change in the API.
But I could solve that providing a patch in spack.

Reply:
Solved with introduced patches

Reply:
Indeed! 👍